### PR TITLE
Enum case

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Currently, debugging task is hard because Timbers uses number instead of name. The type system takes an int to represent a task, which seems also useless. So here I'm trying to solve both problem by adding an enum.

It seems to me that our API does not allow the user to give directly a number to execute, so there should be no problem here if we just change the type

Tested on my phone, because it allows me to understand logcat more easily !

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 1]
```diff
@@ -75,43 +75,45 @@
...
-    public static final int TASK_TYPE_COUNT_MODELS = 41;
-    public static final int TASK_TYPE_DELETE_MODEL = 42;
-    public static final int TASK_TYPE_DELETE_FIELD = 43;
-    public static final int TASK_TYPE_REPOSITION_FIELD = 44;
-    public static final int TASK_TYPE_ADD_FIELD = 45;
-    public static final int TASK_TYPE_CHANGE_SORT_FIELD = 46;
-    public static final int TASK_TYPE_SAVE_MODEL = 47;
-    public static final int TASK_TYPE_FIND_EMPTY_CARDS = 48;
-    public static final int TASK_TYPE_CHECK_CARD_SELECTION = 49;
+    public enum TASK_TYPE {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:78]: Can you please define the integer values explicitly - I think we'll be fine here, but I've been burned a few times by someone adding enum values in the middle, or at the start, which changes all the values, and can lead to mismatches if they're persisted anywhere.

**Comment by Arthur-Milchior** [comment]
Done; but I don't see how it helps as we can't create an enum element from the number and it seems this is used only internally

**Comment by david-allison-1** [comment]
> Done; but I don't see how it helps as we can't create an enum element from the number and it seems this is used only internally

I think you're right.... in .NET you can easily cast an int to an enum and vice-versa (even with invalid values for the enum). Java seems much more constrained in this fashion. Apologies for wasting your time.

**Comment by Arthur-Milchior** [comment]
Reverted to original commit

**Comment by Arthur-Milchior** [comment]
As expected, most of my PR had to be rebased. My merge commit is branch, need to recreate it entirely here

**Pull request merged**